block.py

```diff
@@ -1,33 +1,34 @@
 import hashlib
 
 from merkle_tree import MerkleTree
-import pickle
+from ipv8.messaging.serialization import default_serializer
+from ipv8.messaging.lazy_payload import VariablePayload
+from ipv8.messaging.serialization import default_serializer
 
+class Block(VariablePayload):
+    format_list = ['varlenHutf8', 'varlenHutf8', 'varlenH-list']
+    names = ['previous_hash', 'merkle_hash', 'transactions']
 
-class Block:
-    def __init__(self, previous_hash, transactions=None):
-        self.previous_hash = previous_hash
-        self.transactions = transactions if transactions is not None else []
+    def __init__(self, previous_hash: str = '', merkle_hash: str = '', transactions = []):
+        self.previous_hash = str(previous_hash)
+        self.transactions = transactions if len(transactions) > 0 else []
         self.merkle_tree = MerkleTree()
+        self.merkle_hash = merkle_hash if merkle_hash != '' else ''
 
     def add_transaction(self, transaction):
-        self.transactions.append(transaction)
-        serialized_data = self.serialize(transaction)
-        transaction_hash = hashlib.sha256(serialized_data).hexdigest()
+        # We serialize the transaction before appending it to the array
+        # because there seems to be an ipv8 problem when it tries to serialize the whole array
+        serialized_transaction = default_serializer.pack_serializable(transaction)
```

Comment:
here we need to exclude some field like ttl from transaction before serializing, but I will fix it with my pr